### PR TITLE
rewrites changelog generator to support zenhub-based releases

### PR DESCRIPTION
**Pull request by zkat**
Fixes: https://github.com/NuGet/Home/issues/9085

### CODE REVIEW

File: ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj [hunk 1]
```diff
@@ -9,9 +9,10 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>ChangelogGenerator</RootNamespace>
     <AssemblyName>ChangelogGenerator</AssemblyName>
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
```
Comment (nkolev92) [inline ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: Use SDK based proj + PackageReference?
😢

Comment (zkat) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: Migrated. Hopefully I did that right?

Comment (nkolev92) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: Yup, looks correct.

File: ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                        issueFixed = false;
+                    }
+
+                    if (label.Name == "RegressionDuringThisVersion")
+                    {
+                        regressionDuringThisVersion = true;
+                        hidden = true;
+                    }
+
+                    if (label.Name == "Category:Engineering")
```
Comment (nkolev92) [inline ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.cs]: Can we extract these labels to a file with statics?
Specifically my concern is what happens when my label cleanup ideas get adopted :D 
Would make it easier to fix up the tool.

File: ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj [hunk 3]
```diff
@@ -1,71 +1,35 @@
...
+    <TargetFramework>net472</TargetFramework>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <PublishUrl>publish\</PublishUrl>
+    <Install>true</Install>
+    <InstallFrom>Disk</InstallFrom>
+    <UpdateEnabled>false</UpdateEnabled>
+    <UpdateMode>Foreground</UpdateMode>
+    <UpdateInterval>7</UpdateInterval>
+    <UpdateIntervalUnits>Days</UpdateIntervalUnits>
+    <UpdatePeriodically>false</UpdatePeriodically>
```
Comment (nkolev92) [inline ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: What does this do?

Comment (zkat) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: I have NO idea. It was added by one of the migrators???

File: ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.cs (hunk 2 above)
Comment (zkat) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.cs]: This is a good plan. I'll do that.

File: ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj (hunk 3 above)
Comment (nkolev92) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: Weird. :D 

I think you can safely remove them, but I don't care if you don't do it :)

Comment (rrelyea) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: those were mostly clickonce update properties. it is possible to publish .NET Framework projects with clickonce, and they will look for updates and keep updated. chrome used to use this as their underlying update mechanism on windows, i believe. yes, fine to remove or not.

File: ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj [hunk 4]
```diff
@@ -1,71 +1,29 @@
...
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{95B80A75-AF6E-424A-880A-3EFA4FDF26F4}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>ChangelogGenerator</RootNamespace>
-    <AssemblyName>ChangelogGenerator</AssemblyName>
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net472</TargetFramework>
```
[the remark this answers is not in the document]
Comment (rrelyea) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: (long term, probably good to do...no reason to block progress on that now)

[the remark this answers is not in the document]
Comment (zkat) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: done!

File: ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj (hunk 3 above)
Comment (zkat) [reply ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.csproj]: removed :)

**Pull request merged**